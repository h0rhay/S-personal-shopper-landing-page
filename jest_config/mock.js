export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
