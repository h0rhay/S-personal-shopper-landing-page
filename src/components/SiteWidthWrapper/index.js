import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const SiteWidthContainer = styled.div`
  margin: 0 auto;
  max-width: 90vw;
  width: 70rem;
`;

const SiteWidthWrapper = ({ children, ...delegated }) => (
  <SiteWidthContainer className="site_width_wrap" {...delegated}>
    {children}
  </SiteWidthContainer>
);

SiteWidthWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default SiteWidthWrapper;
