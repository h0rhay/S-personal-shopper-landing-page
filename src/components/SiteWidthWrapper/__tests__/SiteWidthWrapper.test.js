import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import SiteWidthWrapper from '../index';

const props = {
  children: <div>test</div>,
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <SiteWidthWrapper {...props} />
    </ThemeProvider>
  );

describe('SiteWidthWrapper component', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme, props });
    expect(container).toMatchSnapshot();
  });
});
