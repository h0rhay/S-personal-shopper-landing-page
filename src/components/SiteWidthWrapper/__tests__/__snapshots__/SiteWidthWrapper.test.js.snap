// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteWidthWrapper component should match screenshot 1`] = `
<div>
  <div
    class="sc-bdvvtL kiJDep site_width_wrap"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
